### ISSUE TITLE
Name overlaps

### ISSUE THREAD

**Issue by asinghvi17** [JuliaGeo/JuliaGeo.jl#1]
e.g `Rasters.reproject` v/s `GeometryOps.reproject`, or similar overlaps.

Similarly for ArchGDAL and LibGEOS, etc.

What should we do about these?   One option is to define a `JuliaGeo.reproject` that dispatches based on the type of the first argument (at least with Rasters it's fairly obvious).  Depending on the number of intersections, it might make sense to add some methods stubs to GeometryOpsCore.

**Comment by evetion** [comment]
But do either packages also export those names? Your proposal is similar to https://github.com/JuliaGeo/GeoInterface.jl/issues/141 no?

**Comment by rafaqz** [comment]
It would work for Rasters but GO doesn't own any of the arguments. Putting it in GOCore would help with that

**Comment by asinghvi17** [comment]
I guess it kind of is, if we're doing this metapackage thing?  But I don't think this is a very widespread problem, it's really just these two and a few others that intersect, not the entire API or something.  

The main thing there was ArchGDAL, LibGEOS and GeometryOps.  But if we're standardizing on GeometryOps then we don't necessarily need to create a blessed implementation.   Rasters only overlaps with GeometryOps on three names of consequence:

```julia
julia> intersect(names(Rasters), names(GeometryOps))
5-element Vector{Symbol}:
 :Planar
 :Spherical
 :coverage
 :rebuild
 :reproject
```

Planar and Spherical are shared types so that's fine.  It's just `rebuild`, `coverage`, and `reproject` that are issues.  

Rasters should use GeometryOps `coverage` ideally, since they do the same thing.  We can explore dispatches for that.

Reproject can also be moved to GeometryOpsCore and Rasters can define a single dispatch on that.

Rebuild is more internal to GeometryOps, we don't necessarily need to export it I think.

**Comment by rafaqz** [comment]
Yeah rebuild is just my go to name for an extensible object reconstuctor. Only the Rasters one is exported.

`coverage` and`reproject` can extend GOCore

**Comment by rafaqz** [comment]
Yeah of course GO can accept them all and work out the dispatch, Extent/Extent just goes to extents.
